programs/bpf_loader/src/lib.rs

```diff
@@ -1058,30 +1056,38 @@ impl Executor for BpfExecutor {
                         }
                     };
                     stable_log::program_failure(&log_collector, &program_id, &error);
-                    return Err(error);
+                    Err(error)
                 }
+                _ => Ok(()),
             }
-            execute_time.stop();
-        }
+        };
+        execute_time.stop();
+
         let mut deserialize_time = Measure::start("deserialize");
-        let keyed_accounts = invoke_context.get_keyed_accounts()?;
-        deserialize_parameters(
-            &loader_id,
-            &keyed_accounts[first_instruction_account + 1..],
-            parameter_bytes.as_slice(),
-            &account_lengths,
-            invoke_context
-                .feature_set
-                .is_active(&do_support_realloc::id()),
-        )?;
+        let execute_or_deserialize_result = execuction_result.and_then(|_| {
+            let keyed_accounts = invoke_context.get_keyed_accounts()?;
+            deserialize_parameters(
+                &loader_id,
+                &keyed_accounts[first_instruction_account + 1..],
+                parameter_bytes.as_slice(),
+                &account_lengths,
+                invoke_context
+                    .feature_set
+                    .is_active(&do_support_realloc::id()),
+            )
+        });
         deserialize_time.stop();
+
+        // Update the timings
         let timings = &mut invoke_context.timings;
         timings.serialize_us = timings.serialize_us.saturating_add(serialize_time.as_us());
         timings.create_vm_us = timings.create_vm_us.saturating_add(create_vm_time.as_us());
         timings.execute_us = timings.execute_us.saturating_add(execute_time.as_us());
         timings.deserialize_us = timings
             .deserialize_us
             .saturating_add(deserialize_time.as_us());
+
+        execute_or_deserialize_result?;
         stable_log::program_success(&log_collector, &program_id);
```

Comment:
```suggestion
        if execute_or_deserialize_result.is_ok() {
            stable_log::program_success(&log_collector, &program_id);
        }
```

Might make future modifications in this area a little less error prone than the early return

Reply:
would still need to return an error here, the suggestion above always returns `Ok(())` if I'm reading that right

Reply:
Of course can't fix the recommendation without deleting the comment :man_facepalming: 

```
-        execute_or_deserialize_result?;
-        stable_log::program_success(&log_collector, &program_id);
-        Ok(())
+        if execute_or_deserialize_result.is_ok() {
+            stable_log::program_success(&log_collector, &program_id);
+        }
+        execute_or_deserialize_result
```